doc: update genrest script for latest kconfiglib

dbkinder opened the pull request:
The genrest.py script (used to generate config documentation from the
Kconfig files) broke when upgrading to the latest kconfiglib (10.9.1).
Copied the latest genrest.py script from the Zephyr project (where this
script and processing was developed) and things work again.

Also update requirements.txt to match the kconfiglib version verified to
work.

fixes: #1387

Signed-off-by: David B. Kinder <david.b.kinder@intel.com>

gvancuts commented:
Thanks @dbkinder , I'll test this now.

You;ll need to change the `fixes:` to `Tracked-On` for our CI system to be happy though

dbkinder commented:
DO NOT MERGE until #1389 is addressed or all CI builds that do a `make docs` will fail.

gvancuts commented:
@dbkinder , I have thinking and thinkering with this a bit this morning. We should augment this PR to also add a check for the `kconfiglib` in your `doc/scripts/show-versions.py` script.

Thinking about it some more, it would actually be better for the `show-versions.py` script to gather all requirements captured in the file and print out the versions that are actually installed... so I came up with this alternative script:
```
#!/usr/bin/env python3
#
# Copyright (c) 2018, Intel Corporation
#
# SPDX-License-Identifier: BSD-3-Clause
#
# Show installed versions of doc building tools

import os.path
import sys
import pkg_resources
import subprocess

class color:
   PURPLE = '\033[95m'
   CYAN = '\033[96m'
   DARKCYAN = '\033[36m'
   BLUE = '\033[94m'
   GREEN = '\033[92m'
   YELLOW = '\033[93m'
   RED = '\033[91m'
   BOLD = '\033[1m'
   UNDERLINE = '\033[4m'
   END = '\033[0m'

# Check all requirements listed in requirements.txt and print out version installed (if any)
print (color.BOLD + color.BLUE + "Listing out dependencies found on your sytem..." + color.END)

rf = open(os.path.join(sys.path[0], "requirements.txt"),"r")

for reqs in pkg_resources.parse_requirements(rf):
    try:
        ver = pkg_resources.get_distribution(reqs.project_name).version
        print (color.DARKCYAN + reqs.project_name + " version: " + ver + color.END)
    except:
        print (color.RED + color.BOLD + reqs.project_name + " is missing." + color.END +
                " (Hint: install all dependencies with " + color.YELLOW +
                "\"pip3 install --user -r requirements.txt\"" + color.END + ")")

rf.close()

# Print out the version of Doxygen (not installed via pip3)
print (color.DARKCYAN + "doxygen version: " + subprocess.check_output(["doxygen", "-v"]).decode("utf-8") + color.END)
```

This is an example of the output (minus the colouring and formatting):
```
Listing out dependencies found on your sytem...
breathe version: 4.9.1
sphinx version: 1.7.7
docutils is missing. (Hint: install all dependencies with "pip3 install --user -r requirements.txt")
sphinx-rtd-theme version: 0.4.0
kconfiglib version: 10.9.2
doxygen version: 1.8.13

```

What do you think?

dbkinder commented:
Thanks. Fixed a misspelling (sytem), removed colors (other than the error message), and added some indenting and padding to make it a bit easier to read.  (The blue and dark cyan were hard to read on a black background terminal.)

gvancuts commented:
Is there a real risk that `pkg_resources` returns a different version than the one being used? We can keep your version if you think the risk is real. My starting point when going down that path was really that I wanted to be able to extract from the `requirements.txt` file what the deps were and print out the versions for each (as opposed to maintaining two lists in different files). It seems cleaner but keeping these two lists in sync isn't a whole lot of work either so I'm fine either way.

dbkinder commented:
I'm OK with using pkg_resources, just was an observation.  And I like getting the list of dependencies from requirements.txt, so it works for me.
